src/main/groovy/life/qbic/db/MariaDBManager.groovy

```diff
@@ -37,68 +37,90 @@ class MariaDBManager implements IQueryService {
     this.dataSource = dataSource.getSource()
   }
 
-  void addNewLocation(String sampleId, Location location) {
+  void addNewLocation(String sampleId, Location location) throws IllegalArgumentException{
     Connection connection = Objects.requireNonNull(dataSource.getConnection(), "Connection must " +
             "not be null.")
     Sql sql = new Sql(connection)
-    HttpResponse response = HttpResponse.ok(location);
+
+    //validate location
+    int locationId = getLocationIdFromName(location.getName(), sql);
+    if (locationId == -1) {
+      String msg = "Location " + location.getName() + " was not found."
+      log.error(msg)
+      throw new IllegalArgumentException(msg)
+    }
+
+    int personId = getPersonIdFromEmail(location.getResponsibleEmail(), sql)
+    if (personId == -1) {
+      String msg = "User with email " + location.getResponsibleEmail() + " was not found."
+      log.error(msg)
+      throw new IllegalArgumentException(msg)
+    }
```

Comment:
Since you moved the "withTransaction" part further below, are connections closed in the case the two calls here fail?

Reply:
good question

Reply:
6d52d47